src/sgame/sg_votes.h

```diff
@@ -0,0 +1,94 @@
+/*
+===========================================================================
+
+Unvanquished GPL Source Code
+Copyright (C) 2022 Unvanquished Developers
+
+This file is part of the Unvanquished GPL Source Code (Unvanquished Source Code).
+
+Unvanquished is free software; you can redistribute it
+and/or modify it under the terms of the GNU General Public License as
+published by the Free Software Foundation; either version 2 of the License,
+or (at your option) any later version.
+
+Unvanquished is distributed in the hope that it will be
+useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Unvanquished; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+===========================================================================
+*/
+#include "sg_local.h"
+
+enum VoteType
+{
+	V_ANY,     // Vote can be called either as a team vote or public vote
+	V_TEAM,    // Team vote only
+	V_PUBLIC,  // Public vote only
+};
+
+enum VoteTarget
+{
+	T_NONE,    // No arg required
+	T_PLAYER,  // Arg is a player
+	T_OTHER,   // Arg is a random string
+};
+
+enum VoteOptions
+{
+	VOTE_ALWAYS,   // default
+	VOTE_BEFORE,   // within the first N minutes
+	VOTE_AFTER,    // not within the first N minutes
+	VOTE_REMAIN,   // within N/2 minutes before the timelimit
+	VOTE_NO_AUTO,  // don't automatically vote 'yes'
+};
+
+using VoteHandler =
+	std::function<bool( gentity_t* ent, team_t team, std::string& cmd, std::string& arg,
+                        std::string& reason, std::string& name, int clientNum, int id )>;
+
+struct VoteDefinition
```

Comment:
If we're going to do the F macro it would be nice to have default values for optional fields, so we can omit some of them like you said, and make it easier to avoid undefined behavior.

Reply:
It's been removed after feedback on IRC.

Reply:
nvm, my changes were lost after rebase :sob:

Reply:
Added back